### ISSUE TITLE
Pipeline metadata

### ISSUE THREAD

**Issue by simonaubertbd** [amphi-ai/amphi-etl#56]
Hello,

it would be nice, for data governance, to have some metadata on our pipelines :
-author
-copyright
-title
-key words
-a long description
-ability to add a link for documentation
-version

Best regards,

Simon

**Comment by simonaubertbd** [comment]
We can even imagine a tool "amphinception" who parses the pipelines to retrieve the metadata. I love when a tool is able to analyze its own assets.

**Comment by tgourdel** [comment]
Definitely, I'm currently weighing in the best way to achieve this. For now, everything configurable about a pipeline is through components that you can drag and drop. I'm thinking it could be the same for metadata (would be easier for me and maybe for the users as well). The only thing I don't like with this approach is it also clutters the pipeline view. In any case, the metadata would be present in the pipeline DSL in JSON, which can be read by Amphi:
![image](https://github.com/user-attachments/assets/decd3d7c-2b5d-4725-a91f-80170c29bb85)

**Comment by simonaubertbd** [comment]
Oh, and of course, it should be exported to generated python code (with an option to not export)
